src/python/tests/gff3/test_extract_annotation.py

```diff
@@ -58,15 +66,217 @@
+
+
+@pytest.mark.parametrize(
+    "parent_type, parent_id, child_id, expected",
+    [
+        ("gene", "geneA", "mrnA", does_not_raise()),
+        ("bad_type", "geneA", "mrnA", raises(KeyError)),
+        ("gene", "geneB", "mrnA", raises(MissingParentError)),
+    ],
+)
+@pytest.mark.dependency(name="add_parent", depends=["add_feature"])
+def test_add_parent(parent_type, parent_id, child_id, expected):
+    """Tests the `FunctionaAnnotation.add_parent_link()` method.
+
+    Add a parent feature, and then add a parent link.
+
+    Args:
+        parent_type: Type for the parent sequence feature.
+        parent_id: ID for the parent sequence feature.
+        child_id: ID for the child sequence feature.
+        expected: What exception is expected to be raised, if any.
+
+    """
+    annot = FunctionalAnnotations()
+    parent = SeqFeature(type="gene", id="geneA")
+    annot.add_feature(parent, "gene")
+
+    with expected:
+        annot.add_parent_link(parent_type, parent_id, child_id)
+
+
+@pytest.mark.parametrize(
+    "in_parent_type, in_parent_id, in_child_id, out_parent_type, out_child_id, expected",
+    [
+        ("gene", "geneA", "mrnA", "gene", "mrnA", does_not_raise()),
+        ("gene", "geneA", "mrnA", "bad_type", "mrnA", raises(KeyError)),
+        ("gene", "geneA", "mrnA", "gene", "mrnB", raises(MissingParentError)),
+    ],
+)
+@pytest.mark.dependency(name="get_parent", depends=["add_parent"])
+def test_get_parent(in_parent_type, in_parent_id, in_child_id, out_parent_type, out_child_id, expected):
+    """Tests the `FunctionaAnnotation.get_parent()` method.
+
+    Args:
+        in_parent_type: Type for the parent sequence feature.
+        in_parent_id: ID for the parent sequence feature.
+        in_child_id: ID for the child sequence feature.
+        out_parent_type: Type for the parent stored in the functional annotation.
+        out_child_id: ID for the child stored.
+        expected: What exception is expected to be raised, if any.
+
+    """
+    annot = FunctionalAnnotations()
+    parent = SeqFeature(type=in_parent_type, id=in_parent_id)
+    annot.add_feature(parent, "gene")
+    annot.add_feature(SeqFeature(type="mRNA", id=in_child_id), feat_type="transcript", parent_id=in_parent_id)
+
+    with expected:
+        out_parent = annot.get_parent(out_parent_type, out_child_id)
+        assert out_parent == in_parent_id
+
+
+@pytest.mark.parametrize(
+    "child_type, child_id, out_parent_id, expected",
+    [
+        ("transcript", "mrna_A", "gene_A", does_not_raise()),
+        pytest.param("bad_type", "mrna_A", "gene_A", raises(KeyError), id="Child type does not exist"),
+        pytest.param("gene", "gene_A", None, raises(AnnotationError), id="Feature ID already loaded"),
+        pytest.param(
+            "gene", "gene_B", "gene_A", raises(AnnotationError), id="Cannot add a gene child of a gene"
+        ),
+    ],
+)
+@pytest.mark.dependency(name="add_feature_fail", depends=["add_feature", "get_parent"])
```

Comment:
```suggestion
@pytest.mark.dependency(name="add_feature_fail", depends=["add_feature", "get_parent_link"])
```

Reply:
That's `get_parent`, not `add_parent`, so no change here?

Reply:
Cannot recall why I made this suggestion :see_no_evil: Happy to ignore it :+1: